packages/synapse-react-client/src/utils/hooks/useCookiePreferences.ts

```diff
@@ -0,0 +1,45 @@
+import { useCallback } from 'react'
+import { atom, useAtomValue, useSetAtom } from 'jotai'
+
+export const getCurrentCookiePreferences = () => {
+  const prefs = localStorage.getItem(COOKIES_AGREEMENT_LOCALSTORAGE_KEY)
+  const allowNone: CookiePreference = {
+    functionalAllowed: false,
+    analyticsAllowed: false,
+  }
+  return prefs != null ? (JSON.parse(prefs) as CookiePreference) : allowNone
```

Comment:
optionally consider catching exceptions thrown by `JSON.parse` and falling back to `allowNone` in that case as well